pkg/custom-plugins/plugins.go

```diff
@@ -0,0 +1,117 @@
+package customplugins
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// Specs is a list of plugin specs.
+type Specs []Spec
+
+// Spec is a plugin spec and configuration.
+type Spec struct {
+	// Name is the name of the plugin.
+	// Does not allow whitespace characters.
+	Name string `json:"name"`
+
+	// StatePlugin represents the jobs to run for /states API.
+	StatePlugin *Plugin `json:"state_plugin,omitempty"`
+	// EventPlugin represents the jobs to run for /events API.
+	EventPlugin *Plugin `json:"event_plugin,omitempty"`
+
+	// DryRun is set to true to allow non-zero exit code on the script
+	// useful for dry runs.
+	DryRun bool `json:"dry_run"`
+
+	// Timeout is the timeout for the script execution.
+	Timeout metav1.Duration `json:"timeout"`
+
+	// Interval is the interval for the script execution.
+	// If zero, it runs only once.
+	Interval metav1.Duration `json:"interval"`
+}
+
+// Plugin represents a plugin spec.
+type Plugin struct {
+	// Script is the script to run for this job.
+	// Assumed to be base64 encoded.
+	Script string `json:"script"`
+}
```

Comment:
Defining as a struct, in case we want to support container-based executions in the future:

```diff
--- a/pkg/custom-plugins/plugins.go
+++ b/pkg/custom-plugins/plugins.go
@@ -41,6 +41,10 @@ type Plugin struct {
        // Script is the script to run for this job.
        // Assumed to be base64 encoded.
        Script string `json:"script"`
+
+       Envs           map[string]string `json:"envs"`
+       ContainerImage string            `json:"container_image"`
+       ContainerArgs  []string          `json:"container_args"`
 }
```